data/scripting/starting_conditions.lua

```diff
@@ -0,0 +1,27 @@
+include "scripting/coroutine.lua"
+include "scripting/ui.lua"
+
+function find_ocean_fields(how_many)
```

Comment:
I think we should shift this to C++ to make it a lot faster, just like we did for the Territorial and Wood Gnome starting conditions. See functions around `size_t Map::count_all_fields_excluding_caps`.